"You can replace malloc with your own implementation, but it has to be a
*working* implementation."  Details:

http://cygwin.com/ml/cygwin/2009-11/msg00114.html

--- origsrc/graphviz-2.36.0/lib/vmalloc/features/vmalloc	2014-01-11 16:59:44.000000000 -0600
+++ src/graphviz-2.36.0/lib/vmalloc/features/vmalloc	2014-02-09 20:56:30.200892800 -0600
@@ -19,17 +19,18 @@ lib mstats
 hdr alloca
 hdr dlfcn
 
-std	malloc note{ stuck with standard malloc }end noexecute{
+std	malloc note{ stuck with standard malloc }end execute{
 	_BEGIN_EXTERNS_
 	extern void _exit _ARG_((int));
 	extern char* strdup _ARG_((const char*));
+	static int stuck_with_malloc = 0;
 	#if _STD_
-	char* malloc(unsigned n) { _exit(0); return 0; }
+	char* malloc(unsigned n) { stuck_with_malloc = 1; return 0; }
 	#else
-	char* malloc(n) unsigned n; { _exit(0); return 0; }
+	char* malloc(n) unsigned n; { stuck_with_malloc = 1; return 0; }
 	#endif
 	_END_EXTERNS_
-	main() { strdup("yo"); _exit(1); }
+	main() { strdup("yo"); _exit(stuck_with_malloc); }
 }end
 
 lib alloca note{ alloca exists }end compile{
@@ -48,6 +49,7 @@ mal alloca note{ alloca is based on mall
 	#if _hdr_dlfcn
 	#include <dlfcn.h>
 	volatile int testing_alloca = 0;
+	static int based_on_malloc = 1;
 	#endif
 	#if _STD_
 	void* malloc(unsigned int size)
@@ -60,7 +62,7 @@ mal alloca note{ alloca is based on mall
 			return ((void *(*)(unsigned int)) dlsym(RTLD_NEXT, "malloc"))(size);
 		}
 	#endif
-		exit(0);
+		based_on_malloc = 0;
 		return 0;
 	}
 	main()
@@ -72,7 +74,7 @@ mal alloca note{ alloca is based on mall
 	#if _hdr_dlfcn
 		testing_alloca = 0;
 	#endif
-		return 1;
+		return based_on_malloc;
 	}
 }end
 
